zulipterminal/ui_tools/boxes.py

```diff
@@ -88,6 +102,37 @@ def private_box_view(self, button: Any=None, email: str='',
         ]
         self.focus_position = self.FOCUS_CONTAINER_MESSAGE
 
+        # Typing status is sent in regular intervals to limit the number of
+        # notifications sent. Idleness should also prompt a notification.
+        # Refer to https://zulip.com/api/set-typing-status for the protocol
+        # on typing notifications sent by clients.
+        TYPING_STARTED_WAIT_PERIOD = 10
+        TYPING_STOPPED_WAIT_PERIOD = 5
+
+        start_period_delta = timedelta(seconds=TYPING_STARTED_WAIT_PERIOD)
+        stop_period_delta = timedelta(seconds=TYPING_STOPPED_WAIT_PERIOD)
+
+        def on_type_send_status(edit: object, new_edit_text: str) -> None:
+            if new_edit_text:
+                self.last_key_update = datetime.now()
+                if self.last_key_update > self.send_next_typing_update:
+                    self.model.send_typing_status_by_user_ids(
+                                self.recipient_user_ids, status='start')
+                    self.send_next_typing_update += start_period_delta
+                    # Initiate tracker function only if it isn't already
+                    # initiated.
+                    if not self.idle_status_tracking:
+                        self.idle_status_tracking = True
+                        track_idleness_and_update_status()
+
+        @asynch
+        def track_idleness_and_update_status() -> None:
+            while datetime.now() < self.last_key_update + stop_period_delta:
+                continue
```

Comment:
I was a little wary when I noticed that you'd removed the sleep import - we really shouldn't use a busy-loop. At a minimum we should use some kind of sleep in here, and while this is maybe more complex due to the time updates, I wonder if a `threading.Event` might work here?

Reply:
I just switched to using sleep again, since we have very few threads now.